mungegithub/github/timer.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package github
+
+import "time"
+
+// Timer tracks issues that needs to be retriggered later
+type Timer struct {
+	ids map[int]time.Time
+	now func() time.Time
+}
+
+// NewTimer creates a new timer
+func NewTimer() *Timer {
+	return &Timer{
+		ids: make(map[int]time.Time),
+		now: time.Now,
+	}
+}
+
+// GetReached returns the list of issues whose timer has been recched
+func (timer *Timer) GetReached() []int {
+	reached := []int{}
+
+	for id, date := range timer.ids {
+		if timer.now().After(date) {
+			delete(timer.ids, id)
```

Comment:
This is likely fine (I just started reviewing) but it makes me feel icky that `Get` has in implicit `Remove`.